### PR TITLE
Add python plotting helper

### PR DESCRIPTION
**Pull request by HeinrichHartmann**

### CODE REVIEW

**Comment by HeinrichHartmann** [comment]
Travis CI found some problems with python2.7 that I was not aware of. Nice!
@cody271 Can we also add a test run python 3.6 to Travis? Not sure how to do that.

File: README.md [hunk 1]
```diff
@@ -36,3 +36,17 @@ Then use the following command to install the python library.
 
 This repository contains lua bindings for libcircllhist, which are generated by `make`,
 and will be installed by `make install` to the directory `/usr/local/share/lua/5.1/`.
+
+## FAQ
+
+### After the installation, libcircllhist.so can not be found by my program / the python bindings. How can I fix this?
```
Comment (cody271) [inline README.md]: If this has been a problem for users, we could also embed the native library inside the Python module..

File: src/generatePythonFiles.sh [hunk 2]
```diff
@@ -22,5 +22,11 @@ for path in [ # Search for libcircllhist.so
      pass
 if not C:
    # let dlopen throw it's error
+   print("""
```
Comment (cody271) [inline src/generatePythonFiles.sh]: Should rethrow combined exception instead of printing to console here

File: src/generatePythonFiles.sh [hunk 3]
```diff
@@ -22,5 +22,11 @@ for path in [ # Search for libcircllhist.so
...
+
+libcircllhist.so was not found on your system.
+Please install libcircllhist from: https://github.com/circonus-labs/libcircllhist/
+
+   """)
    ffi.dlopen("libcircllhist.so")
 EOF
```
Comment (cody271) [inline src/generatePythonFiles.sh]: Also this file seems to have some strange indenting, 2 and **3** spaces! Should be all 4 spaces.

File: src/python/circllhist/wrapper.py [hunk 4]
```diff
@@ -169,3 +177,56 @@ def compress_mbe(self, mbe):
         - return the compressed histogram as new value
         """
         return Circllhist(ffi.C.hist_compress_mbe(self._h, mbe), gc=True)
```
Comment (cody271) [inline src/python/circllhist/wrapper.py]: Make sure the MBE is always an integer here with `int(mbe)`

File: src/python/circllhist/wrapper.py [hunk 5]
```diff
@@ -169,3 +177,56 @@ def compress_mbe(self, mbe):
...
+
+    def plot(self, **kwargs):
+        """
+        Plot histogram using matplotlib.
+        Depends on matplotlib being available.
+        Retruns an axes object.
```
Comment (cody271) [inline src/python/circllhist/wrapper.py]: "**Returns** an axes object."

File: src/python/circllhist/wrapper.py [hunk 6]
```diff
@@ -169,3 +177,56 @@ def compress_mbe(self, mbe):
...
+
+        Hint: Use H.plot().set_xlim(x_min, x_max) to adjust value range.
+
+        mbe-compression is applied before plotting so squash bins which are deemed to small.
```
Comment (cody271) [inline src/python/circllhist/wrapper.py]: "mbe-compression is applied before plotting **to** squash bins which are deemed **too** small." ?

File: src/generatePythonFiles.sh (hunk 2 above)
Comment (cody271) [reply src/generatePythonFiles.sh]:
```python
if not C:
    try:
        ffi.dlopen("libcircllhist.so")
    except Exception as e:
        raise Exception(str(e) + """

libcircllhist.so was not found on your system.
Please install libcircllhist from: https://github.com/circonus-labs/libcircllhist/

        """)
```

File: README.md (hunk 1 above)
Comment (HeinrichHartmann) [reply README.md]: Yeah. I was not sure how to do that. We would need different libs for x64/x32 and potentially multiple OSes. I did not want to go into this. Maybe this a problem for the C layer to address. We do have EL7/Ubuntu packages AFAIK.

File: src/generatePythonFiles.sh (hunk 2 above)
Comment (HeinrichHartmann) [reply src/generatePythonFiles.sh]: We effectively do that in the next line.

**Comment by HeinrichHartmann** [comment]
Addressed review comments. Travis passed. Merging this one.

**Pull request merged**